src/pages/DetailsPage/DetailsPage.spec.js

```diff
@@ -50,11 +52,41 @@ describe('DetailsPage', () => {
           },
         }}
         onClickDetailsBack={onClickDetailsBack}
+        onToggleFavorite={jest.fn()}
+        isFavorite={false}
       />
     )
 
-    const back = screen.getByRole('button')
+    const back = screen.getByTestId('back')
```

Comment:
You should _always_ prefer to use `getByRole` and try to avoid `getByTestId` whenever possible.
You can read about that in the [testing-library docs](https://testing-library.com/docs/queries/bytestid)

> In the spirit of the guiding principles, it is recommended to use this only after the other queries don't work for your use case. Using data-testid attributes do not resemble how your software is used and should be avoided if possible. That said, they are way better than querying based on DOM structure or styling css class names. Learn more about data-testids from the blog post "Making your UI tests resilient to change"